Data/SessionService.cs

```diff
@@ -10,19 +10,56 @@ public sealed class SessionService
     public SessionService(IDbConnection dbConnection) =>
         _dbConnection = dbConnection;
 
-    public IReadOnlyList<Session> GetSessions() =>
-        _dbConnection.Query<Session, Author, Session>(
-            @"SELECT s.id AS sessionId, s.when, a.id AS authorId, a.name, a.surname 
-                FROM session s
-                JOIN author a
-                    ON s.chair_id = a.id",
-            (s, a) =>
+    public IReadOnlyList<Session> GetSessions() {
+
+        var paperDictionary = new Dictionary<int, Paper>();
+
+        PaperService t = new PaperService(_dbConnection);
+
+        foreach (var paper in t.GetPapers().OrderBy(a => a.Id)) {
+            paperDictionary.Add(paper.Id, paper);
+        }
+ 
+        var sessionDictionary = new Dictionary<int, Session>();
+
+        var result = _dbConnection.Query<Session, Author, Lecture, Author, int, Session>(
+            @"SELECT s.id as sessionId, s.when AS when,
+                a2.id AS authorId, a2.name AS name, a2.surname AS surname,
+                COALESCE(l.id, -1) AS lectureId, COALESCE(l.when, '2022-01-12') AS when,
+                COALESCE(a1.id, -1) AS authorId, COALESCE(a1.name, '-') AS name, coalesce(a1.surname, '-') AS surname,
+                coalesce(l.paper_id, -1) As paperId
```

Comment:
Po co te wszystkie `COALESCE`? Jedyny rzeczywiście przydatny to ten z `paperId`, a i go da się uniknąć używając typu `int?` zamiast `int`. Poniższy kod działa dokładnie tak samo, a jest trochę prostszy i nie używa magicznej wartości `-1`:

```csharp
var result = _dbConnection.Query<Session, Author, Lecture, Author, int?, Session>(
    @"SELECT s.id as sessionId, s.when AS when,
        a2.id AS authorId, a2.name AS name, a2.surname AS surname,
        l.id AS lectureId, l.when AS when,
        a1.id AS authorId, a1.name AS name, a1.surname AS surname,
        l.paper_id As paperId
    FROM session s LEFT JOIN lecture l on s.id = l.session_id
        LEFT JOIN author a1 ON l.speaker_id = a1.id LEFT JOIN author a2 ON s.chair_id = a2.id",

    (s, a2, l, a1, paperId) =>
    {
        Session? session;

        if (!sessionDictionary.TryGetValue(s.Id, out session))
        {
            session = s;
            sessionDictionary.Add(s.Id, session);
        }

        session.Chair = a2;

        if (l is not null)
        {
            if (paperId is not null)
            {
                l.Paper = paperDictionary[paperId.Value];
            }
            l.Speaker = a1;
            l.Session = session;
            session.Lectures.Add(l);
        }

        return session;
    },
    splitOn: "authorId,lectureId,authorId,paperId");
```